samples/src/JustSaying.Sample.Middleware/Middlewares/PollyJustSayingMiddleware.cs

```diff
@@ -0,0 +1,40 @@
+﻿using JustSaying.Messaging.Middleware;
+using JustSaying.Sample.Middleware.Exceptions;
+using Polly;
+using Serilog;
+
+namespace JustSaying.Sample.Middleware.Middlewares;
+
+/// <summary>
+/// A middleware that will wrap the handling of a message in the provided Polly policy.
+/// </summary>
+public class PollyJustSayingMiddleware : MiddlewareBase<HandleMessageContext, bool>
+{
+    private readonly AsyncPolicy _policy;
+
+    public PollyJustSayingMiddleware()
+    {
+        _policy = CreateMessageRetryPolicy();
+    }
+
+    protected override async Task<bool> RunInnerAsync(HandleMessageContext context, Func<CancellationToken, Task<bool>> func, CancellationToken stoppingToken)
+    {
+        Log.Information("[{MiddlewareName}] Started {MessageType}", nameof(PollyJustSayingMiddleware), context.Message.GetType().Name);
+
+        try
+        {
+            return await _policy.ExecuteAsync(() => func(stoppingToken));
```

Comment:
With Polly it's best practice to also allow Polly's internals access to the `CancellationToken`.

```suggestion
            return await _policy.ExecuteAsync(() => func(stoppingToken), stoppingToken);
```

Reply:
Hmm this doesn't compile and doesn't look like there's an overload for passing in a CT 🤔

Reply:
It was there, I just looked badly